packages/medusa/src/api/routes/store/products/list-products.ts

```diff
@@ -194,8 +196,18 @@ export default async (req, res) => {
   const cartService: CartService = req.scope.resolve("cartService")
   const regionService: RegionService = req.scope.resolve("regionService")
 
+  const featureFlagRouter: FlagRouter = req.scope.resolve("featureFlagRouter")
+
   const validated = await validator(StoreGetProductsParams, req.query)
 
+  if (featureFlagRouter.isFeatureEnabled(PublishableAPIKeysFeatureFlag.key)) {
+    if (req.publishableApiKeyScopes?.sales_channel_id.length) {
+      validated.sales_channel_id =
+        validated.sales_channel_id ||
+        req.publishableApiKeyScopes.sales_channel_id
+    }
+  }
+
```

Comment:
thought: if the user provides a `sales_channel_id` query param we should validate that that id is one of the sales channels the publishable key gives access to. This could be done in a middleware.

Reply:
It's implemented this way now because I wanted to allow that request params, if present, to override PKs.
If we think of PKs as a fixed filter that you only have to define once, does it makes sense that the sales channels defined by the PK in the request header are only relevant if a sales channel isn't passed as a param?

On the other hand, If we think that PKs define a set of available resources your proposal with additional middleware sounds good!

Reply:
My take would be that we should consider the PK a way for the creator to say: "all requests with this key should only be allowed to access these resources" 

Over time (i.e. in the future, not now) I would even argue that we should deprecate usage of the storefront API without a PK.

Reply:
Sounds good - I added additional `validateSalesChannelParam` middleware that checks if passed SC is "allowed" by the PK from the header